src/main/java/bookstore/service/UserServiceImpl.java

```diff
@@ -4,25 +4,39 @@
 import bookstore.dto.user.UserResponseDto;
 import bookstore.exception.RegistrationException;
 import bookstore.mapper.UserMapper;
+import bookstore.model.Role;
 import bookstore.model.User;
+import bookstore.repository.RoleRepository;
 import bookstore.repository.UserRepository;
 import jakarta.transaction.Transactional;
 import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 @Transactional
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
     private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         if (userRepository.existsByEmail(requestDto.getEmail())) {
             throw new RegistrationException("Email already exists: " + requestDto.getEmail());
         }
+        Role.RoleName roleName = Role.RoleName.USER;
+        Role defaultRole = roleRepository.findByRole(roleName);
+        if (defaultRole == null) {
+            defaultRole = new Role();
+            defaultRole.setRole(Role.RoleName.USER);
```

Comment:
you shouldn't create such a default role manually. Please add it via liquibase and throw and exception in case role is absent